### DISCUSSION TITLE
bluetoot gnss receiver

### DISCUSSION THREAD

**Discussion by goviforestale** [opengisch/QField#3814] [Bugs]
after the last update, (2.6.1) on android 12,  i can't connect my bluetoot gnss, i try and i get the message "location service tourned off"

**Comment by nirvn** [comment]
@goviforestale , could you attach a screencast here that shows the issue? Please show in your screencast that Bluetooth connection is on.

**Comment by goviforestale** [comment]
![Screenshot_2023-01-20-12-36-54-491_ch opengis qfield](https://user-images.githubusercontent.com/123162789/213687100-f15c01db-8f4c-4bc3-b528-cab7c738c068.jpg)


if i try with bluetooth gps app it works fine


![Screenshot_2023-01-20-12-31-57-680_googoo android btgps](https://user-images.githubusercontent.com/123162789/213687583-42ac5faf-9176-422d-9aaa-e5093a07663e.jpg)

**Comment by nirvn** [comment]
@goviforestale , I can replicate the scan error **when I turn of location services** in my Android settings. This leads me to think your phone has location services turned off too, which triggers the message you are reporting above in Qt. This behavior by the Qt library insures that your intent to turn off location discovery on your phone is respected.

Solution: please open your Android settings and search for the keyword 'location' and select 'Location services'. In there, you'll likely see a toggle that's switched off. Turn it on, et voila.

**Reply by goviforestale** [reply]
I'm sorry, but I'm still having problems
I have two bluetooth receivers:
GNS3000 and Navilock BT-821G

with navilook I connect it with the android positioning on, then I can turn it off and qfiel keeps giving me the position.
if i do the same with gns3000, when i turn off the android positioning qfiel loses the position (blue dot), so I also have the doubt that in this case qfiel reads the position from the internal device and not from the external receiver.
before the update of qfied even with gns3000 I had turned off the "positioning of androd"
now qfiel 2.6.3

**Comment by goviforestale** [comment]
I removed version 2.6.3 and installed the old version 2.6.0.
With 2.6.0 I can also connect with gns3000 and it works fine.

????

**Comment by goviforestale** [comment]
ok, i did a lot of tests, from version 2.6 and later my receiver (gns3000) only connects the first time and then it is not possible to reconnect it.
I also tested with a different smartphone (also with another receiver, always of the same model).
Now i'm using 2.5.7 version and i can disconnect and reconnect it without problems.
Could it be because the pairing code is generated each time and is not the same as for other bluetooth devices?

then with version 2.5.7 the connection takes place even with the android location service off, and it is a great advantage to save battery of the android device

**Reply by nirvn** [reply]
@goviforestale , can you please provide a video recording showing a successful first connect than failing reconnects on 2.6? Thanks.

**Reply by goviforestale** [reply]
https://drive.google.com/file/d/1-rmYoFqO287m-thTJb7gKcPTmox30lS8/view?usp=share_link

this time it didn't connect even on the first try

**Comment by goviforestale** [comment]
https://drive.google.com/file/d/1-rmYoFqO287m-thTJb7gKcPTmox30lS8/view?usp=share_link

this time it didn't connect even on the first try

**Reply by modulelele** [reply]
I faced same problem too. and, I solved it.
In my case, I was using the GPS connector app for another mapping app, When I stopped that app and then connected the Qfield to the GPS device, the bug did not occur.
Are you using an app that uses a bluetooth device other than Qfield, such as a compass or satellite count app?

**Reply by goviforestale** [reply]
I try to uninstall the other apps that use gps, I hadn't noticed.
one question, what model of bluetooth gps are you using?
Is the pairing code fixed, for example 0000 or 1234, or is it generated each time by the device, as in the case of the gps I use?

**Reply by goviforestale** [reply]
Nothing to do, it doesn't work for me.
I go back to 2.5.7

**Reply by modulelele** [reply]
I guess I couldn't help you much...Just in case, let me explain our equipment.
No PIN is used for pairing(SSP). 
GPS:GG-1LT
phone:aquos sense7 (android 13)
Qfield:2.7.2

**Reply by nirvn** [reply]
@goviforestale , could you try these APKs: https://github.com/opengisch/QField/pull/4079#issuecomment-1484516616

I'm moderately hopeful it'll improve things for you. Let me know.

**Reply by goviforestale** [reply]
it does not work.
find gns3000 but when I ask for the connection the answer is this
![Screenshot_2023-03-29-07-52-52-118_ch opengis qfield_beta](https://user-images.githubusercontent.com/123162789/228439650-11f4655b-a137-4a70-ad00-521024bde89b.jpg)

with 2.5.7 it connects correctly

**Reply by nirvn** [reply]
@goviforestale , try with these new APKs: https://github.com/opengisch/QField/pull/4079#issuecomment-1484516616 -- this could well be it :)

**Comment by goviforestale** [comment]
I've been trying for half an hour, I turn off, turn on gnss and bluetooth
and now everything seems to work fine
Thank you 😘

Il ven 31 mar 2023, 08:59 Mathieu Pellerin ***@***.***> ha
scritto:

**Reply by mbernasocchi** [reply]
Fantastic news! Happy you could confirm all our testing 🚀 
looking forward to 2.7.3 :)
